mlrose/algorithms.py

```diff
@@ -87,7 +87,10 @@ def hill_climb(problem, max_iters=np.inf, restarts=0, init_state=None,
             problem.find_neighbors()
             next_state = problem.best_neighbor()
             next_fitness = problem.eval_fitness(next_state)
-
+            
+            if curve:
```

Comment:
I understand what you are trying to do here. However, if you make this change in the code, then the fitness curve will include the fitness of state that you never actually move to, which I think is wrong.